pkg/serial/handshake.go

```diff
@@ -200,48 +172,56 @@ func HandshakeServer(ctx context.Context, conn net.Conn) error {
 	}
 
 	log.Debug("server: reading syn")
-	// syn is 3 bytes as that will eventually syn us again if we're offset
-	if n, err := readMultiple(conn, syn); n != 2 || err != nil || syn[0] != flagSyn {
-		var msg string
-		if err != nil {
-			msg = fmt.Sprintf("server: failed to read expected SYN: n=%d, err=%s", n, err)
-		} else if syn[0] != flagSyn {
-			msg = fmt.Sprintf("server: did not receive SYN (read %v bytes): %#x != %#x", n, flagSyn, syn[0])
-			conn.Write([]byte{flagNak})
-		} else {
-			msg = fmt.Sprintf("server: received SYN (read %d) bytes", n)
-		}
-
-		// to aid in debug we always dump the full handhsake
-		log.Debug(msg)
-		log.Debugf("server: read %v bytes: ", n)
-		for i := 0; i < n; i++ {
-			log.Debugf("%#x ", syn[i])
-		}
-		log.Debug("")
+	// syn is 2 bytes as that will eventually syn us again if we're offset
```

Comment:
This comment is no longer correct - if you're sending and reading 2 bytes then there is nothing in this piece of code that will result in eventual window realignment.